superset/jinja_context.py

```diff
@@ -213,6 +165,147 @@ def url_param(
             self.cache_key_wrapper(result)
         return result
 
+    def filter_values(
+        self, column: str, default: Optional[str] = None, remove_filter: bool = True
+    ) -> List[str]:
+        """Gets a values for a particular filter as a list
+
+        This is useful if:
+            - you want to use a filter box to filter a query where the name of filter
+             box column doesn't match the one in the select statement
+            - you want to have the ability for filter inside the main query for speed
+            purposes
+
+        Usage example::
+
+            SELECT action, count(*) as times
+            FROM logs
+            WHERE
+                action in ({{ "'" + "','".join(filter_values('action_type')) + "'" }})
+            GROUP BY action
+
+        :param column: column/filter name to lookup
+        :param default: default value to return if there's no matching columns
+        :param remove_filter:
+            flag indicating if superset should generate a condition for this filter
+        :return: returns a list of filter values
+        """
+        from superset.views.utils import get_form_data
+
+        form_data, _ = get_form_data()
+        convert_legacy_filters_into_adhoc(form_data)
+        merge_extra_filters(form_data)
+
+        return_val = [
+            comparator
+            for filter in form_data.get("adhoc_filters", [])
+            for comparator in (
+                filter["comparator"]
+                if isinstance(filter["comparator"], list)
+                else [filter["comparator"]]
+            )
+            if (
+                filter.get("expressionType") == "SIMPLE"
+                and filter.get("clause") == "WHERE"
+                and filter.get("subject") == column
+                and filter.get("comparator")
+            )
+        ]
+
+        if return_val:
+            if remove_filter:
+                self.cache_key_wrapper({"filter_removed_by_vt": column})
```

Comment:
I believe all filters are converted into adhocs filters that have a unique id. IMO it would be optimal if we would collect the removed filter ids in a dedicated `dict` on `ExtraCache` called `removed_filters` and then make it possible to check if a filter has been removed or not without looping through all extra cache keys.

Reply:
I'll add the `removed_filters` however I can't use the `filterOptionName` because in the sqla/models.py it's not made available..

Reply:
ended up using a set of column names
it simplified the check done in sqla/models.py